consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/TransitionProtocolScheduleTest.java

```diff
@@ -159,52 +155,45 @@ public void getByBlockHeader_returnsTimestampScheduleIfPresent() {
   }
 
   @Test
-  public void getByBlockNumber_returnsTimestampScheduleIfPresent() {
-    when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.of(blockHeader));
-    when(timestampSchedule.getByBlockHeader(blockHeader)).thenReturn(mock(ProtocolSpec.class));
-
-    assertThat(transitionProtocolSchedule.getByBlockNumber(BLOCK_NUMBER)).isNotNull();
-
-    verify(timestampSchedule).getByBlockHeader(blockHeader);
-    verifyNoMergeScheduleInteractions();
-  }
-
-  @Test
-  public void getByBlockNumber_delegatesToPreMergeScheduleWhenBlockNotFound() {
-    when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.empty());
+  public void getByBlockHeader_delegatesToPreMergeScheduleWhenBlockNotFound() {
     when(mergeContext.isPostMerge()).thenReturn(false);
 
-    transitionProtocolSchedule.getByBlockNumber(BLOCK_NUMBER);
+    transitionProtocolSchedule.getByBlockHeaderForBws(blockHeader);
 
     verifyPreMergeProtocolScheduleReturned();
   }
 
   @Test
-  public void getByBlockNumber_delegatesToPostMergeScheduleWhenBlockNotFound() {
-    when(blockchain.getBlockHeader(BLOCK_NUMBER)).thenReturn(Optional.empty());
+  public void getByBlockHeader_delegatesToPostMergeScheduleWhenBlockNotFound() {
+    when(blockHeader.getNumber()).thenReturn(BLOCK_NUMBER);
+    when(blockHeader.getDifficulty()).thenReturn(Difficulty.ZERO);
+    when(blockchain.getTotalDifficultyByHash(any())).thenReturn(Optional.of(TTD));
     when(mergeContext.isPostMerge()).thenReturn(true);
 
-    transitionProtocolSchedule.getByBlockNumber(BLOCK_NUMBER);
+    transitionProtocolSchedule.getByBlockHeaderForBws(blockHeader);
 
     verifyPostMergeProtocolScheduleReturned();
   }
 
   @Test
-  public void getByBlockNumber_delegatesToPostMergeScheduleWhenTimestampScheduleDoesNotExist() {
+  public void getByBlockHeader_delegatesToPostMergeScheduleWhenTimestampScheduleDoesNotExist() {
```

Comment:
Test names should be updated from getByBlockHeader_... -> getByBlockHeaderForBws_...

Reply:
Changed these tests to use getByBlockHeader so we have test coverage for that method